src/diskquota.c

```diff
@@ -1006,13 +1005,14 @@ create_monitor_db_table(void)
 		HOLD_INTERRUPTS();
 		EmitErrorReport();
 		FlushErrorState();
+		if (!SPI_context()) SPI_restore_connection();
```

Comment:
What happens if the `SPI_connect_and_check` function above fails?

Reply:
I've created the ADBDEV-7316 ticket